internal/pkg/workspace/workspace.go

```diff
@@ -519,68 +498,49 @@ func (ws *Workspace) writeSummary(appName string) error {
 	return ws.fs.WriteFile(summaryPath, serializedWorkspaceSummary, 0644)
 }
 
-func (ws *Workspace) pipelinesDirPath() (string, error) {
-	copilotPath, err := ws.copilotDirPath()
-	if err != nil {
-		return "", err
-	}
-	return filepath.Join(copilotPath, pipelinesDirName), nil
+func (ws *Workspace) pipelinesDirPath() string {
+	return filepath.Join(ws.copilotDirAbs, pipelinesDirName)
 }
 
-func (ws *Workspace) summaryPath() (string, error) {
-	copilotPath, err := ws.copilotDirPath()
-	if err != nil {
-		return "", err
-	}
-	workspaceSummaryPath := filepath.Join(copilotPath, SummaryFileName)
-	return workspaceSummaryPath, nil
+func (ws *Workspace) summaryPath() string {
+	return filepath.Join(ws.copilotDirAbs, SummaryFileName)
 }
 
-func (ws *Workspace) createCopilotDir() error {
+func (ws *Workspace) createCopilotDir() (string, error) {
 	// First check to see if a manifest directory already exists
 	existingWorkspace, _ := ws.copilotDirPath()
 	if existingWorkspace != "" {
-		return nil
+		return existingWorkspace, nil
+	}
+	if err := ws.fs.Mkdir(CopilotDirName, 0755); err != nil {
+		return "", err
 	}
-	return ws.fs.Mkdir(CopilotDirName, 0755)
+	return filepath.Join(ws.workingDirAbs, CopilotDirName), nil
```

Comment:
is this different than before?

Reply:
Do you mean if `filepath.Join(ws.workingDirAbs, CopilotDirName)` really always the copilot directory path? (correct me if im wrong!)

If so, I think it's so (that it is the copilot dir path)!

`ws.fs.Mkdir(CopilotDirName, 0755)` is intermediately  `ws.fs.Mkdir("copilot", 0755)`, which will create a `copilot` dir under the current work dir - since `"copilot"` is not an absolute path. Since `ws.workingDirAbs` always gets its value from `os.Getwd()`, its value should be correct - so it checks out 🤔 !

Reply:
oh I get it now I am so dumb, I don't know what I was thinking. 👍 totally makes sense

For some reason, I kept reading the old code and thinking that `ws.fs.Mkdir()` returned a `(string, error)` rather than just error 🤦